### DISCUSSION TITLE
Rich dependency failure or PEBCAK?

### DISCUSSION THREAD

**Discussion by brianjmurrell** [rpm-software-management/rpm#3457] [Q&A] [answered]
If I have a BR (line only broken below for clarity, it's all one long line the actual spec file):

```
BuildRequires:  (pkgconfig(ffnvcodec) >= 12.1.14.0 or
                 (pkgconfig(ffnvcodec) >= 12.0.16.1 and pkgconfig(ffnvcodec) < 12.1) or
                 (pkgconfig(ffnvcodec) >= 11.1.5.2 and pkgconfig(ffnvcodec < 12.0) or
                 (pkgconfig(ffnvcodec) >= 11.0.10.3 and pkgconfig(ffnvcodec < 11.1) or
                 (pkgconfig(ffnvcodec) >= 8.1.24.15 and pkgconfig(ffnvcodec < 8.2))
```

and I have installed `nv-codec-headers-11.1.5.2-1.el9.noarch` which provides:

```
nv-codec-headers = 11.1.5.2-1.el9
pkgconfig(ffnvcodec) = 11.1.5.2
```
shouldn't the above BR: allow the build to happen?

I'm actually getting:

```
Error: 
 Problem: nothing provides requested (pkgconfig(ffnvcodec) >= 12.1.14.0 or (pkgconfig(ffnvcodec) >= 12.0.16.1 and pkgconfig(ffnvcodec) < 12.1) or (pkgconfig(ffnvcodec) >= 11.1.5.2 and pkgconfig(ffnvcodec < 12.0) or (pkgconfig(ffnvcodec) >= 11.0.10.3 and pkgconfig(ffnvcodec < 11.1) or (pkgconfig(ffnvcodec) >= 8.1.24.15 and pkgconfig(ffnvcodec < 8.2))
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

As an aside, is there any way to actually break long lines like the above in a specfile?  Can I use `\` for example?

**Comment by mlschroe** [comment, marked as answer]
Your parens don't match up.  There are a couple of `)` missing for the sub-expressions. I'm not sure why this could be parsed at all.

Btw, you likely want to use `with` instead of `and`, as both the `>=` and the `<` should be provided by the same package.

**Reply by brianjmurrell** [reply]
> Your parens don't match up. There are a couple of ) missing for the sub-expressions. I'm not sure why this could be parsed at all.

Doh!  Yes, I see.

> Btw, you likely want to use with instead of and, as both the >= and the < should be provided by the same package.

Yes, I did do that subsequently to creating this topic.  Thanks for the head's up though.

One thing not answered from the original questions:

> is there any way to actually break long lines like the above in a specfile?
